cypress/e2e/ui/Settings/Application-Settings/tenant.cy.js

```diff
@@ -0,0 +1,634 @@
+  nameInputFieldId: 'name',
+  descriptionInputFieldId: 'description',
+
+  // Config options
+  configToolbarButton: 'Configuration',
+  editTenantConfigOption: 'Edit this item',
+  addProjectConfigOption: 'Add Project to this Tenant',
+  deleteItemConfigOption: 'Delete this item',
+  manageQuotasConfigOption: 'Manage Quotas',
+  addChildTenantConfigOption: 'Add child Tenant to this Tenant',
+
+  // Quota Descriptions
+  allocatedStorageQuota: 'Allocated Storage in GB',
+  allocatedVmQuota: 'Allocated Number of Virtual Machines',
+
+  // Buttons
+  saveButton: 'Save',
+  cancelButton: 'Cancel',
+  resetButton: 'Reset',
+  addButton: 'Add',
+
+  // Flash message types
+  flashTypeSuccess: 'success',
+  flashTypeWarning: 'warning',
+  flashTypeError: 'error',
+
+  // Flash message text snippets
+  flashMessageOperationCanceled: 'cancel',
+  flashMessageOperationReset: 'reset',
+  flashMessageSaved: 'saved',
+  flashMessageAdded: 'added',
+  flashMessageCantDelete: 'cannot delete',
+  browserAlertDeleteConfirmText: 'delete',
+  nameAlreadyTakenError: 'taken',
+};
+
+const {
+  // Component route url
+  componentRouteUrl,
+
+  // Menu options
+  settingsMenuOption,
+  appSettingsMenuOption,
+
+  // Accordion items
+  accessControlAccordionItem,
+  accessControlAccordionItemId,
+  manageIQRegionAccordItem,
+  tenantsAccordionItem,
+
+  // Field values
+  formHeaderFragment,
+  initialParentTenantName,
+  initialParentTenantDescription,
+  editedTenantNameValue,
+  editedDescriptionValue,
+  projectNameValue,
+  initialChildTenantName,
+  initialChildTenantDescription,
+
+  // Element ids
+  nameInputFieldId,
+  descriptionInputFieldId,
+
+  // Config options
+  configToolbarButton,
+  editTenantConfigOption,
+  addProjectConfigOption,
+  deleteItemConfigOption,
+  manageQuotasConfigOption,
+  addChildTenantConfigOption,
+
+  // Quota Descriptions
+  allocatedStorageQuota,
+  allocatedVmQuota,
+
+  // Buttons
+  saveButton,
+  cancelButton,
+  resetButton,
+  addButton,
+
+  // Flash message types
+  flashTypeSuccess,
+  flashTypeWarning,
+  flashTypeError,
+
+  // Flash message text snippets
+  flashMessageOperationCanceled,
+  flashMessageOperationReset,
+  flashMessageSaved,
+  flashMessageAdded,
+  flashMessageCantDelete,
+  browserAlertDeleteConfirmText,
+  nameAlreadyTakenError,
+} = textConstants;
+
+function confirmUiNavigation(callback) {
+  cy.url()
+    ?.then((url) => {
+      // Ensures navigation to Settings -> Application-Settings in the UI
+      if (!url?.includes(componentRouteUrl)) {
+        // Navigate to Settings -> Application-Settings before looking out for parent Tenants edited during test
+        cy.menu(settingsMenuOption, appSettingsMenuOption);
+      }
+    })
+    .then(() => {
+      callback();
+    });
+}
+
+function cancelForm(
+  assertFlashMessage = true,
+  flashMessageSnippet = flashMessageOperationCanceled
+) {
+  cy.getFormFooterButtonByType(cancelButton).click();
+  if (assertFlashMessage) {
```

Comment:
It looks like assertFlashMessage = true was copied from saveForm but it's never an argument to cancelForm.  I think you can remove the assignment and if conditional here.